src/CarbonAware.WebApi/test/integrationTests/mocks/WattTimeDataSourceMocker.cs

```diff
@@ -0,0 +1,53 @@
+﻿using CarbonAware.DataSources.Configuration;
+using CarbonAware.Tools.WattTimeClient;
+using CarbonAware.Tools.WattTimeClient.Configuration;
+using CarbonAware.Tools.WattTimeClient.Model;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using WireMock.Server;
+
+namespace CarbonAware.WebApi.IntegrationTests;
+public class WattTimeDataSourceMocker : IDataSourceMocker
+{
+    protected WireMockServer _server;
+    private readonly object _dataSource = DataSourceType.WattTime;
+
+    internal WattTimeDataSourceMocker()
+    {
+        _server = WireMockServer.Start();
+        WattTimeServerMocks.WattTimeServerSetupMocks(_server);
+    }
+
+    public void SetupDataMock(DateTime start, DateTime end, string location)
+    {
+        GridEmissionDataPoint newDataPoint = WattTimeServerMocks.GetDefaultEmissionsDataPoint();
+        DateTimeOffset newTime = new(start);
+        newDataPoint.PointTime = newTime;
+
+        WattTimeServerMocks.SetupDataMock(_server, new List<GridEmissionDataPoint> { newDataPoint });
+    }
+
+    public WebApplicationFactory<Program> overrideWebAppFactory(WebApplicationFactory<Program> factory)
+    {
+        return factory.WithWebHostBuilder(builder =>
+        {
+            builder.ConfigureServices(services =>
+            {
+                services.Configure<CarbonAwareVariablesConfiguration>(configOpt =>
```

Comment:
after some testing + debugging, it turns out this doesn't actually override the config for `CarbonIntensityDataSource`, so am adding the Env. approach back after discussing it with @bderusha -- can sync tomorrow offline

Reply:
Yep. Going to delete these lines, they are useless